pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -0,0 +1,164 @@
+import logging
+import os
+import re
+import shutil
+from collections import namedtuple
+from datetime import datetime
+from pathlib import Path
+import traceback
+from typing import Dict, Iterable, List, Optional, Tuple
+
+import click
+import yaml
+from doozerlib.assembly import AssemblyTypes
+from pyartcd import constants, exectools
+from pyartcd.cli import cli, click_coroutine, pass_runtime
+from pyartcd.record import parse_record_log
+from pyartcd.runtime import Runtime
+from pyartcd.util import (get_assembly_type, get_release_name,
+                          isolate_el_version_in_release, load_group_config, load_releases_config)
+from semver import VersionInfo
+
+PlashetBuildResult = namedtuple("PlashetBuildResult", ("repo_name", "local_dir", "remote_url"))
+
+
+class BuildMicroShiftPipeline:
+    """ Rebase and build MicroShift for an assembly """
+
+    def __init__(self, runtime: Runtime, group: str, assembly: str,
+                 ocp_build_data_url: str, logger: Optional[logging.Logger] = None):
+        self.runtime = runtime
+        self.group = group
+        self.assembly = assembly
+        self._logger = logger or runtime.logger
+        self._slack_client = self.runtime.new_slack_client()
+
+        # determines OCP version
+        match = re.fullmatch(r"openshift-(\d+).(\d+)", group)
+        if not match:
+            raise ValueError(f"Invalid group name: {group}")
+        self._ocp_version = (int(match[1]), int(match[2]))
+
+        # sets environment variables for Doozer
+        self._doozer_env_vars = os.environ.copy()
+        self._doozer_env_vars["DOOZER_WORKING_DIR"] = str(self.runtime.working_dir / "doozer-working")
```

Comment:
Isn't this going to complain?
```
TypeError: unsupported operand type(s) for /: 'str' and 'str'
```
IIRC, `runtime.workdir` is a string

Reply:
It is of type `pathlib.Path`.